lib_collections/models.py

```diff
@@ -217,6 +217,263 @@ class RelatedCollectionPagePlacement(Orderable, models.Model):
     )
 
 
+class ExternalService(models.Model):
+    """
+    Class for link to an external service in a Collection Page.
+    """
+    MENU_OPTIONS = [
+        (1, "LUNA"),
+        (2, "BTAA"),
+    ]
+    service = models.IntegerField(
+        choices=MENU_OPTIONS, default=1, help_text='Choose an external service'
+    )
+    identifier = models.CharField(
+        max_length=255, help_text="identifier to retrieve object from service"
+    )
+
+    panels = [
+        FieldPanel('service'),
+        FieldPanel('identifier'),
+    ]
+
+    class Meta:
+        abstract = True
+
+
+class CollectionPageExternalService(Orderable, ExternalService):
+    """
+    Intermediate class for links to external services in a Collection
+    Page. (needed to create an InlinePanel)
+    """
+    page = ParentalKey(
+        'lib_collections.CollectionPage', related_name="col_external_service"
+    )
+
+
+class ObjectMetadata(models.Model):
+    """
+    Class for metadata fields to display in search results.
+    """
+    MENU_OPTIONS = [
+        (1, "go to a results page for the selected item"),
+        (2, "link to a related item in the collection"),
+    ]
+    edm_field_label = models.CharField(max_length=255, blank=True)
+    multiple_values = models.BooleanField(
+        default=False, help_text='Are there multiple values within the field?'
+    )
+    hotlinked = models.BooleanField(
+        default=False,
+        help_text='Do you want this label to linked to a unique browse view?'
+    )
+    link_target = models.IntegerField(
+        choices=MENU_OPTIONS,
+        default=1,
+        help_text='How do you want the link to behave? (Required for hotlinked)'
+    )
+
+    panels = [
+        FieldPanel('edm_field_label'),
+        FieldPanel('hotlinked'),
+        FieldPanel('multiple_values'),
+        FieldPanel('link_target'),
+    ]
+
+    class Meta:
+        abstract = True
+
+
+class CollectionPageObjectMetadata(Orderable, ObjectMetadata):
+    """
+    Intermediate class for metadata fields within a Collection Page
+    result.  (needed to create an InlinePanel)
+    """
+    page = ParentalKey(
+        'lib_collections.CollectionPage', related_name="col_obj_metadata"
+    )
+
+
+class CResult(models.Model):
+    """
+    Class for search results within a Collection Page.
+    """
+    field_label = models.CharField(max_length=255, blank=True)
+    field_identifier = models.CharField(
+        max_length=255, blank=True, help_text="EDM/IIIF field identifier"
+    )
+
+    panels = [
+        FieldPanel('field_label'),
+        FieldPanel('field_identifier'),
+    ]
+
+    class Meta:
+        abstract = True
+
+
+class CollectionPageResult(Orderable, CResult):
+    """
+    Intermediate class for results within a Collection Page.  
```

Comment:
Still have a flake8 W291: trailing whitespace error here.

I wonder why your linter isn't catching these. When I pull this up in vim, it highlights these for me. There might be a config you can change in emacs or something like that

Reply:
Sorry about that; I'll look into it and get back to you.  Could be a config setting that needs to be tweaked.

Reply:
Weird!  Looks like my `flake8` installation was messed up!  Emacs was reporting "no flake8 errors" but the command line app was actually erroring out!  Sheesh.  I think I've fixed it by reinstalling flake8 through pip.  Checking that now...

Reply:
Now that my `flake8` is working again, it's throwing some warnings about some of the old code.  (For example, it doesn't like some of the bare `except` clauses.)  What do we usually do when that happens?  Leave it alone?  Make `flake8` happy?  Either?

Reply:
Ah yes, I should have caught those. Yes, we should make flake8 happy and test for the actual exception types. It's much better for readability and shows that we understand what it's supposed to be doing. It's also less error prone. I've seen cases where the exception type actually mattered and by catching all exceptions something unexpected happened (and it was tough to troubleshoot because it wasn't throwing the error it was supposed to because we were catching it). It also gives a nice hook for writing unit tests.

Reply:
But yeah, we do have some old bare exceptions laying around. I usually fix them when I stumble across them.